exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/vendors/DataTypeProvider.kt

```diff
@@ -127,6 +127,10 @@ abstract class DataTypeProvider {
     open fun jsonBType(): String =
         throw UnsupportedByDialectException("This vendor does not support binary JSON data type", currentDialect)
 
+    /** Data type for arrays with no specified size or element type, used only as types of [QueryParameter]s for PostgreSQL and H2.
+     * An array with no element type cannot be used for storing arrays in a column in either PostgreSQL or H2. */
+    abstract fun untypedAndUnsizedArrayType(): String
+
```

Comment:
Let's go by the majority here, since only 2 dialects are implemented. Please swap this with the following and remove all exceptions from unsupported dialect files:
```kt
open fun untypedAndUnsizedArrayType(): String =
        throw UnsupportedByDialectException("This vendor does not support array data type", currentDialect)
```